### ISSUE TITLE
Getting nan-loss or CUDA_ERROR_ILLEGAL_ADDRESS after upgrading to 2.5.0 

### ISSUE THREAD

**Issue by stefan-falk** [tensorflow/tensorflow#50326]
**System information**
- OS Platform and Distribution: Ubuntu 16.04
- TensorFlow version: v2.5.0-rc3-213-ga4dfb8d1a71 
- Python version: 3.8
- CUDA/cuDNN version: 11.1
- GPU model and memory: 4x GeFoce 1080 GTX
- NVIDIA driver version: 465.19.01

**Describe the current behavior**

After upgrading to 2.5.0 none of my experiments work anymore. 

To be precise, ~any training where I am using LSTM-layers~ (turns out the LSTM-layer itself cannot be the problem) one of my models always ends in either a `nan`-loss or a `CUDA_ERROR_ILLEGAL_ADDRESS`. This issue does not come right at the beginning. I am usually able to get a few hundred batches through before the problem occurs.

I don't know if these are two related or separate issues though. All I can say is that this problem did not exist before.

**Describe the expected behavior**

Work as previously in 2.4.1

**Other info / logs**

```
2021-06-18 11:33:42.734501: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
2021-06-18 11:33:42.734547: F tensorflow/core/common_runtime/device/device_event_mgr.cc:221] Unexpected Event status: 1
Fatal Python error: Aborted

Thread 0x00007f14d7afd700 (most recent call first):
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/multiprocessing/pool.py", line 576 in _handle_results
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 892 in run
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 912 in _bootstrap

Thread 0x00007f1780ff9700 (most recent call first):
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/multiprocessing/pool.py", line 528 in _handle_tasks
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 892 in run
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 912 in _bootstrap

Thread 0x00007f17817fa700 (most recent call first):
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/selectors.py", line 416 in select
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/multiprocessing/connection.py", line 936 in wait
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/multiprocessing/pool.py", line 499 in _wait_for_updates
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/multiprocessing/pool.py", line 519 in _handle_workers
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 892 in run
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/home/sfalk/miniconda3/envs/asr2/lib/python3.9/threading.py", line 912 in _bootstrap

Thread 0x00007f1781ffb700 (most recent call first):
  File "/home/sfa2021-06-18 11:33:42.736286: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
lk/miniconda2021-06-18 11:33:42.736311: E tensorflow/stream_executor/cuda/cuda_driver.cc:1085] could not wait stream on event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
3/envs/asr2/lib/p2021-06-18 11:33:42.736331: E tensorflow/stream_executor/cuda/cuda_event.cc:29] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS: an illegal memory access was encountered
ython3.9/2021-06-18 11:33:42.736358: F tensorflow/core/common_runtime/device/device_event_mgr.cc:221] Unexpected Event status: 1
multiprocessing/pool.py", line 114 in workerAborted (core dumped)
```

**Comment by ymodak** [comment]
You may want to upgrade your cuda version to 11.2 if using pre built TF 2.5 binary.
See tested builds https://www.tensorflow.org/install/source#gpu

**Comment by stefan-falk** [comment]
Oh, I didn't even check. I don't know why but something made me thing I'd need 11.1. I really, really hope that this is the problem because I am not at all in the mood for hunting one additional issue at this point 😆

**Comment by stefan-falk** [comment]
@ymodak Unfortunately the problem still exists. I have first removed/purged the NVIDIA components and then installed CUDA 11.2

```
sudo apt purge nvidia-* -y && sudo apt autoremove -y 
sudo apt-get install cuda-11-2 -y
```
I've also installed cuDNN 8.1.0

```
tar -xvf cudnn-11.2-linux-x64-v8.1.0.77.tgz && \
sudo cp -a cuda/include/* /usr/local/cuda/include/ && \
sudo cp -a cuda/lib64/* /usr/local/cuda/lib64/ && \
sudo ldconfig
```

but after 600 or 700 batches the loss turns keeps turning `nan` 😞

**Comment by stefan-falk** [comment]
I have just found this (see [link](https://docs.nvidia.com/deploy/cuda-compatibility/index.html#cuda-application-compatibility)):

![image](https://user-images.githubusercontent.com/43335432/122545211-dde7e500-d02d-11eb-9e28-488257954d5e.png)

But this is weird.. if I install CUDA 11.2 on Ubuntu (`apt-get install cuda-11-2`), it will automatically install the 465.19.01 version of the NVIDIA driver.

I'm confused now .. the table states that these should not be compatible. Am I getting something wrong?

**Comment by ymodak** [comment]
I wonder if this fails if you force run on cpu? Also have you tried with TF 2.5 final version?

**Comment by stefan-falk** [comment]
Hm, I am using this loss function here https://github.com/HawkAaron/warp-transducer - running it on CPU always resulted in a negative lose so idk if I'd have to recompile it.

What do you mean by final version? My pip version is `2.5.0`.

**Comment by ymodak** [comment]
In your issue template I it says 2.5.0-rc3 version. By final I meant 2.5.0 .
Basically 2.5.0 is later than 2.5.0-rc3
https://github.com/tensorflow/tensorflow/tags

**Comment by stefan-falk** [comment]
Hm, I didn't even notice. But I definitely installed `tensorflow==2.5.0`

```
$ pip install tensorflow -U 
Requirement already satisfied: tensorflow in ./miniconda3/envs/asr2/lib/python3.9/site-packages (2.5.0)
Requirement already satisfied: numpy~=1.19.2 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.19.5)
Requirement already satisfied: wheel~=0.35 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (0.36.2)
Requirement already satisfied: flatbuffers~=1.12.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.12)
Requirement already satisfied: six~=1.15.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.15.0)
Requirement already satisfied: astunparse~=1.6.3 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.6.3)
Requirement already satisfied: gast==0.4.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (0.4.0)
Requirement already satisfied: typing-extensions~=3.7.4 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (3.7.4.3)
Requirement already satisfied: tensorboard~=2.5 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (2.5.0)
Requirement already satisfied: google-pasta~=0.2 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (0.2.0)
Requirement already satisfied: keras-nightly~=2.5.0.dev in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (2.5.0.dev2021032900)
Requirement already satisfied: keras-preprocessing~=1.1.2 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.1.2)
Requirement already satisfied: h5py~=3.1.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (3.1.0)
Requirement already satisfied: termcolor~=1.1.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.1.0)
Requirement already satisfied: grpcio~=1.34.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.34.1)
Requirement already satisfied: wrapt~=1.12.1 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (1.12.1)
Requirement already satisfied: opt-einsum~=3.3.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (3.3.0)
Requirement already satisfied: tensorflow-estimator<2.6.0,>=2.5.0rc0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (2.5.0)
Requirement already satisfied: protobuf>=3.9.2 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (3.17.3)
Requirement already satisfied: absl-py~=0.10 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorflow) (0.12.0)
Requirement already satisfied: google-auth<2,>=1.6.3 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (1.31.0)
Requirement already satisfied: markdown>=2.6.8 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (3.3.4)
Requirement already satisfied: requests<3,>=2.21.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (2.25.1)
Requirement already satisfied: werkzeug>=0.11.15 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (2.0.1)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (1.8.0)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (0.4.4)
Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (0.6.1)
Requirement already satisfied: setuptools>=41.0.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from tensorboard~=2.5->tensorflow) (57.0.0)
Requirement already satisfied: rsa<5,>=3.1.4 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (4.7.2)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (4.2.2)
Requirement already satisfied: pyasn1-modules>=0.2.1 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (0.2.8)
Requirement already satisfied: requests-oauthlib>=0.7.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.5->tensorflow) (1.3.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (0.4.8)
Requirement already satisfied: idna<3,>=2.5 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (2.10)
Requirement already satisfied: chardet<5,>=3.0.2 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (4.0.0)
Requirement already satisfied: certifi>=2017.4.17 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (2021.5.30)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (1.26.5)
Requirement already satisfied: oauthlib>=3.0.0 in ./miniconda3/envs/asr2/lib/python3.9/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.5->tensorflow) (3.1.1)
```

Which gives me:

```
$ python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
2021-06-18 20:52:18.648285: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
v2.5.0-rc3-213-ga4dfb8d1a71 2.5.0

**Comment by stefan-falk** [comment]
I let the training run on CPU over the weekend. It's still running as it should. No nan-loss or exception was thrown.

**Comment by UsharaniPagadala** [comment]
@stefan-falk 

Could you please confirm if the issue still persist.Thanks

**Comment by stefan-falk** [comment]
So far I was not able to fix this problem. I'm still getting this error but I am unsure whether it's me or Tensorflow. All I know is that I didn't have this issue in 2.4.1.

**Comment by sanjoy** [comment]
Can you please run the model with the env var `CUDA_LAUNCH_BLOCKING` set to `1` and attach the full logs?

**Comment by stefan-falk** [comment]
@sanjoy If it's alright then I'll try this in the course of the next few days (maybe over the weekend). I'm a bit busy atm and for this I'll have to run a few experiments because the exception does not get thrown every time.

**Comment by sachinprasadhs** [comment]
Hi, Have you tried the above mentioned step, also please try in the latest Tensorflow version and let us know if the issue still persists. Thanks!
